mono/metadata/file-mmap-posix.c

```diff
@@ -489,8 +493,10 @@ mono_mmap_flush (void *mmap_handle, MonoError *error)
 {
 	MmapInstance *h = (MmapInstance *)mmap_handle;
 
+#ifdef HAVE_MSYNC
```

Comment:
Are there systems with mmap but without msync? It is important sometimes to not silently drop semantics. Sometimes better to fail to compile. These ifdefs in general imply a latent untested combinatorial explosion. Not clearly worth filling the minefield.